includes/class-wc-payments-order-success-page.php

```diff
@@ -255,6 +258,59 @@ private function format_addtional_thankyou_order_received_text( string $addition
 		return sprintf( '<div class="woocommerce-info">%s</div>', $additional_text );
 	}
 
+	/**
+	 * Replace the order received text with a failure message when the order status is 'failed'.
+	 *
+	 * @param string $text The original thank you text.
+	 * @return string
+	 */
+	public function replace_order_received_text_for_failed_orders( $text ) {
+		global $wp;
+
+		$order_id = absint( $wp->query_vars['order-received'] );
+		$order    = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return $text;
+		}
+
+		$intent_id      = $order->get_meta( '_intent_id', true );
+		$payment_method = $order->get_payment_method();
+
+		// Strip the gateway ID prefix from the payment method.
+		$payment_method_type = str_replace( WC_Payment_Gateway_WCPay::GATEWAY_ID . '_', '', $payment_method );
+
+		$should_show_failure = false;
+
+		// Check order status first to avoid unnecessary API calls.
+		if ( $order->has_status( Order_Status::FAILED ) ) {
+			$should_show_failure = true;
+		} elseif ( ! empty( $intent_id ) && ! empty( $payment_method_type ) && in_array( $payment_method_type, Payment_Method::REDIRECT_PAYMENT_METHODS, true ) ) {
+			// For redirect-based payment methods that haven't been marked as failed yet, check the intent status.
+			// Add a small delay to allow the intent to be updated.
+			sleep( 1 );
```

Comment:
Without some delay here, the intent hasn't yet been updated with the new status and error.